src/coreclr/jit/lsra.cpp

```diff
@@ -3075,6 +3086,10 @@ regNumber LinearScan::allocateReg(Interval* currentInterval, RefPosition* refPos
     else if (!found)
     {
         found = selector.applySelection(FREE, freeCandidates);
+        if (found)
+        {
+            INTRACK_STATS(updateLsraStat(LsraStat::REGSEL_FREE, refPosition->bbNum));
```

Comment:
This is such a common pattern, would it be useful to have a macro
```
#define INTRACK_STATS_IF(condition,work) if (condition) work;
```
?

and:
```
INTRACK_STATS_IF(found, updateLsraStat(LsraStat::REGSEL_FREE, refPosition->bbNum));
```
or is that too obfuscating?

Reply:
I like the idea.